core/embed/bootloader/main.c

```diff
@@ -312,6 +286,9 @@ void real_jump_to_firmware(void) {
 
   ensure(check_image_model(hdr), "Wrong firmware model");
 
+  ensure(check_firmware_version(hdr->version, sectrue),
```

Comment:
Splat the function in https://github.com/trezor/trezor-firmware/commit/905b1005146edfe3d8d42b0dfee80c032813981f

Regarding the placement changes in https://github.com/trezor/trezor-firmware/commit/8eb2f4f469aa53a3b47b5546cd8f85a295272a40:
I moved the update call after the signature check and only if sig is valid, but before contents check. Reason being that i want the counter updated in case user manages to upload the header correctly, but the firmware update process  is interrupted. cc @matejcik . Once the header is uploaded, user is stripped away from choice of uploading previous firmware version.

The check in main is done before signature check, as i preferred the device with "downgraded" fw to appear empty, as we do will i.e. model checks etc. (placing after the sig check would cause firmware appear as corrupted. Or we would have to add another UI logic which i am trying to avoid). The placement in real_jump_to_firmware mimics the placement in main.

Reply:
Okay, as we discussed in person, the main problem - updating the monotonic counter with a value from an unreliable source - has been fixed. So, I'm okay with it now.